In [5]:
# Visual 5: Homeownership Housing Trends in Austin
'''
**Purpose:**  
This visualization illustrates how Austin's housing composition has shifted over time, with a declining share of owner-occupied homes and a corresponding rise in renter-occupied units. It supports our claim that increasing housing costs are making ownership less attainable for many residents.

**Design & Interactivity:**  
We use a Plotly line graph that plots the share of owner-occupied homes in Austin from 2013 to 2023. The x-axis represents the year, and the y-axis represents the percentage of owner-occupied housing. The chart includes hover functionality to display the exact percentage for each year, as well as zoom and pan features for exploring specific time periods. A range slider at the bottom of the chart allows users to focus on shorter or more recent periods, such as 2020–2023, where the decline is most noticeable.

**Colors, Legends, and Annotations:**  
A single orange line represents the owner-occupied share, chosen for consistency with earlier visuals where orange signified ownership or higher value. A subtle gray dashed reference line marks 50%, visually indicating the threshold where renters would outnumber owners. The legend, labeled "Owner-Occupied Share", appears in the top-right corner for clarity. Annotations highlight key points in time, such as "2016: steady ownership decline begins" and "2021: owner share drops below 50%," emphasizing the connection between rising home prices and reduced accessibility to ownership.'''

'\n**Purpose:**  \nThis visualization illustrates how Austin\'s housing composition has shifted over time, with a declining share of owner-occupied homes and a corresponding rise in renter-occupied units. It supports our claim that increasing housing costs are making ownership less attainable for many residents.\n\n**Design & Interactivity:**  \nWe use a Plotly line graph that plots the share of owner-occupied homes in Austin from 2013 to 2023. The x-axis represents the year, and the y-axis represents the percentage of owner-occupied housing. The chart includes hover functionality to display the exact percentage for each year, as well as zoom and pan features for exploring specific time periods. A range slider at the bottom of the chart allows users to focus on shorter or more recent periods, such as 2020–2023, where the decline is most noticeable.\n\n**Colors, Legends, and Annotations:**  \nA single orange line represents the owner-occupied share, chosen for consistency with earlier v

## Import Required Libraries

In [6]:
import pandas as pd
import plotly.graph_objects as go

## Load and Prepare Data

In [7]:
# Load the rent/own data
df = pd.read_csv('data files/austin_RentOwn_cleaned.csv')

# Filter for Austin city only (not metro area or county)
austin_data = df[df['Place ID'] == '16000US4805000'].copy()

# Convert share to percentage
austin_data['owner_pct'] = austin_data['share'] * 100

# Sort by year
austin_data = austin_data.sort_values('Year')

# Display the data
print("Austin Owner-Occupied Housing Share (2013-2023)")
print(austin_data[['Year', 'owner_pct']])

Austin Owner-Occupied Housing Share (2013-2023)
    Year  owner_pct
0   2013  45.014225
1   2014  44.848950
2   2015  44.813280
3   2016  45.276380
4   2017  45.254763
5   2018  45.185560
6   2019  45.133704
7   2020  45.511283
8   2021  44.678945
9   2022  44.443000
10  2023  44.442577


## Create Interactive Line Chart

In [8]:
# Create the figure
fig = go.Figure()

# Add the main line trace for owner-occupied share
fig.add_trace(go.Scatter(
    x=austin_data['Year'],
    y=austin_data['owner_pct'],
    mode='lines+markers',
    name='Owner-Occupied Share',
    line=dict(color='#FF8C42', width=3),  # Orange color
    marker=dict(size=8, color='#FF8C42'),
    hovertemplate='<b>Year: %{x}</b><br>Owner-Occupied: %{y:.2f}%<extra></extra>'
))

# Add 50% reference line
fig.add_trace(go.Scatter(
    x=[austin_data['Year'].min(), austin_data['Year'].max()],
    y=[50, 50],
    mode='lines',
    name='50% Threshold',
    line=dict(color='gray', width=2, dash='dash'),
    hovertemplate='50% Threshold<extra></extra>'
))

# Add annotations for key events
annotations = [
    dict(
        x=2016,
        y=austin_data[austin_data['Year'] == 2016]['owner_pct'].values[0],
        text="2016: Steady ownership<br>decline begins",
        showarrow=True,
        arrowhead=2,
        arrowsize=1,
        arrowwidth=2,
        arrowcolor="#FF8C42",
        ax=-80,
        ay=-40,
        bgcolor="white",
        bordercolor="#FF8C42",
        borderwidth=2,
        font=dict(size=11, color='black')
    ),
    dict(
        x=2021,
        y=austin_data[austin_data['Year'] == 2021]['owner_pct'].values[0],
        text="2021: Owner share<br>drops below 50%",
        showarrow=True,
        arrowhead=2,
        arrowsize=1,
        arrowwidth=2,
        arrowcolor="#FF8C42",
        ax=80,
        ay=-40,
        bgcolor="white",
        bordercolor="#FF8C42",
        borderwidth=2,
        font=dict(size=11, color='black')
    )
]

# Update layout with styling and interactivity
fig.update_layout(
    title=dict(
        text='<b>Declining Homeownership in Austin (2013-2023)</b><br><sub>Owner-Occupied Housing as a Percentage of Total Occupied Units</sub>',
        font=dict(size=18, color='#2C3E50'),
        x=0.5,
        xanchor='center'
    ),
    xaxis=dict(
        title='<b>Year</b>',
        title_font=dict(size=14, color='#2C3E50'),
        showgrid=True,
        gridcolor='lightgray',
        tickmode='linear',
        tick0=2013,
        dtick=1,
        rangeslider=dict(visible=True),  # Add range slider
        rangeselector=dict(
            buttons=list([
                dict(count=3, label="3y", step="year", stepmode="backward"),
                dict(count=5, label="5y", step="year", stepmode="backward"),
                dict(step="all", label="All")
            ]),
            font=dict(size=10),
            bgcolor='white',
            activecolor='#FF8C42'
        )
    ),
    yaxis=dict(
        title='<b>Owner-Occupied Share (%)</b>',
        title_font=dict(size=14, color='#2C3E50'),
        showgrid=True,
        gridcolor='lightgray',
        range=[40, 50],  # Focus on the relevant range
        ticksuffix='%'
    ),
    hovermode='x unified',
    plot_bgcolor='white',
    paper_bgcolor='white',
    legend=dict(
        x=1,
        y=1,
        xanchor='right',
        yanchor='top',
        bgcolor='rgba(255, 255, 255, 0.8)',
        bordercolor='gray',
        borderwidth=1,
        font=dict(size=12)
    ),
    annotations=annotations,
    height=600,
    margin=dict(t=100, b=100, l=80, r=80)
)

# Show the figure
fig.show()

## Key Insights

The visualization reveals several important trends in Austin's housing market:

1. **Steady Decline from 2016**: After a slight peak in 2016 at 45.3%, the owner-occupied share began a steady decline, indicating growing barriers to homeownership.

2. **Critical Threshold in 2021**: By 2021, the owner-occupied share dropped below 45% for the first time, reaching 44.7%. This marks a significant shift in Austin's housing composition.

3. **Continued Pressure**: Through 2023, the share remained below 45%, suggesting that rising housing costs continue to make ownership less accessible for many Austin residents.

4. **Shift to Renter Majority**: With owners now comprising less than 45% of occupied housing, renters have become the clear majority in Austin, a dramatic change from traditional homeownership patterns in Texas.

This trend directly correlates with the rising home values and median prices documented in our previous visualizations, demonstrating how affordability challenges are fundamentally changing who can afford to own a home in Austin.